crates/experimentation_platform/src/api/experiment_groups/helpers.rs

```diff
@@ -187,3 +208,257 @@ pub fn remove_members(
         .get_result(conn)?;
     Ok(Json(updated_group))
 }
+
+pub fn update_bucket_allocation(
+    experiment: &Experiment,
+    exp_group_buckets: &mut Buckets,
+    exp_traffic_percentage: &TrafficPercentage,
+) -> superposition::Result<()> {
+    let mut current_exp_buckets = vec![];
+    let mut unassigned_buckets = vec![];
+
+    // Separate current exp buckets and unassigned buckets
+    for bucket in exp_group_buckets.iter_mut() {
+        if let Some(buck) = bucket {
+            if experiment.variants.iter().any(|v| v.id == *buck.variant_id) {
+                current_exp_buckets.push(bucket);
+            }
+        } else {
+            unassigned_buckets.push(bucket);
+        }
+    }
+
+    let required_bucket_count =
+        **exp_traffic_percentage as usize * experiment.variants.len();
+    let current_bucket_count = current_exp_buckets.len();
+
```

Comment:
can we rather do this instead, the order of the operation would not matter then
```suggestion
   let bucket_diff = required_bucket_count.abs_diff(current_bucket_count);
```